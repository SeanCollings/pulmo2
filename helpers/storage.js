import { AsyncStorage } from 'react-native';
import { APP_ID } from '../constants/constants';

export const clearAllAsyncStorage = () => {
  AsyncStorage.clear();
};

export const getAsyncData = async (id) => {
  try {
    const value = await AsyncStorage.getItem(`${APP_ID}${id}`);
    if (value !== null) {
      return JSON.parse(value);
    } else {
      return null;
    }
  } catch (error) {
    console.log('Failure retrieving ' + id + ' error: ' + error);
  }
};

export const getMultiAsyncData = async (id, keys) => {
  try {
    const formattedKeys = keys.map((key) => `${APP_ID}${id}${key}`);
    const values = await AsyncStorage.multiGet(formattedKeys);

    if (values && !!values.length) return values;
    else {
      return null;
    }
  } catch (err) {
    console.log('Failure retrieving from all keys with error ' + error);
  }
};

export const storeAsyncData = async (id, value) => {
  try {
    await AsyncStorage.setItem(`${APP_ID}${id}`, JSON.stringify(value));
    return true;
  } catch (error) {
    console.log('Failure saving: ' + error);
    return false;
  }
};

export const removeAsyncData = async (id) => {
  try {
    await AsyncStorage.removeItem(`${APP_ID}${id}`);
  } catch (err) {
    console.log('Failure removing: ' + error);
  }
};
