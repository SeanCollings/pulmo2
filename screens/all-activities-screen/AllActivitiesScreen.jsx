import React, { useEffect, useState, useContext } from 'react';
import { StyleSheet, View, ActivityIndicator } from 'react-native';

import options from './options';
import { useTheme } from '../../hooks/useTheme';
import { HistoryContext } from '../../context/history-context';
import ListScroller from '../../components/ListScroller';
import ActivityMinDetail from '../../components/ActivityMinDetail';

export const allActivitiesScreenOptions = options;

const AllActivitiesScreen = ({ navigation }) => {
  const theme = useTheme();
  const { activities, activitiesUpdated, getSavedActivites } = useContext(
    HistoryContext
  );
  const [allActivites, setAllActivites] = useState(activities);
  const [isLoading, setIsLoading] = useState(true);

  useEffect(() => {
    if (!activities.length && getSavedActivites) {
      getSavedActivites().then(() => {
        setIsLoading(false);
      });
    } else {
      setIsLoading(false);
    }
  }, []);

  useEffect(() => {
    getSavedActivites().then(() => {
      setIsLoading(false);
    });
    const sortedActivites = activities.sort(
      (a, b) => new Date(b.date) - new Date(a.date)
    );

    setAllActivites(sortedActivites);
  }, [activitiesUpdated]);

  return (
    <View style={{ ...styles.container, backgroundColor: theme.BACKGROUND }}>
      {isLoading && (
        <View style={styles.spinner}>
          <ActivityIndicator size="large" color={theme.SECONDARY} />
        </View>
      )}
      {!isLoading && (
        <ListScroller
          data={allActivites}
          uid={'date'}
          RenderItem={ActivityMinDetail}
          navigation={navigation}
          theme={theme}
        />
      )}
    </View>
  );
};

const styles = StyleSheet.create({
  container: { flex: 1 },
  spinner: {
    flex: 1,
    position: 'absolute',
    alignSelf: 'center',
    top: '50%',
    zIndex: 1,
  },
});

export default AllActivitiesScreen;
