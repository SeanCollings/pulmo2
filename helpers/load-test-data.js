import testData, {
  TEST_KEY,
  TEST_TITLE,
  TEST_MUTLIPLE_ID,
  TEST_DATA,
  TEST_PRE_ID,
} from '../data/test-data';
import { getAsyncData, storeAsyncData } from './storage';

export const MESSAGE_UNKNOWN = 'unknown';
export const MESSAGE_PRE_SAVED = 'pre-saved';
export const MESSAGE_DONE = 'done';

export default async (title) => {
  try {
    const selectedData = Object.values(testData).find(
      (data) => data[TEST_TITLE] === title
    );

    if (!selectedData) return MESSAGE_UNKNOWN;

    if (!selectedData[TEST_MUTLIPLE_ID]) {
      const savedData = await getAsyncData(selectedData[TEST_KEY]);

      if (!savedData) {
        console.log('Saving data for:', selectedData[TEST_KEY]);
        await storeAsyncData(selectedData[TEST_KEY], selectedData[TEST_DATA]);
        return MESSAGE_DONE;
      } else {
        return MESSAGE_PRE_SAVED;
      }
    } else {
      const updatedSaves = await selectedData[TEST_DATA].reduce(
        async (acc, data) => {
          const newAcc = await acc;
          const id = selectedData[TEST_MUTLIPLE_ID];
          const preId = selectedData[TEST_PRE_ID] || '';
          const savedData = await getAsyncData(`${preId}${data[id]}`);

          if (!savedData) {
            console.log('Saving data for:', `${preId}${data[id]}`);
            newAcc.push(data[id]);
            await storeAsyncData(`${preId}${data[id]}`, data);
          }
          return newAcc;
        },
        []
      );

      if (!updatedSaves.length) return MESSAGE_PRE_SAVED;

      return `${MESSAGE_DONE} [${updatedSaves.length}]`;
    }
  } catch (err) {
    throw new Error(err);
  }
};
