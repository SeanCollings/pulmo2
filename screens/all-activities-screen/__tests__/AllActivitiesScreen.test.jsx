import React from 'react';
import { act, create } from 'react-test-renderer';
import { cleanup } from 'react-native-testing-library';
import AllActivitiesScreen, {
  allActivitiesScreenOptions,
} from '../AllActivitiesScreen';
import HistoryContextProvider from '../../../context/history-context';
import { getAsyncData, getMultiAsyncData } from '../../../helpers/storage';

jest.useFakeTimers();
jest.mock('react-native/Libraries/Animated/src/NativeAnimatedHelper');
jest.mock('../../../helpers/storage');

describe('AllActivitiesScreen - unit test', () => {
  let tree;
  const navigation = { goBack: () => {} };
  const activityIdArray = [
    '2020-07-26T15:15:43.411Z',
    '2020-06-12T15:15:43.411Z',
    '2020-07-25T15:15:43.411Z',
    '2020-02-22T15:15:43.411Z',
  ];
  const activities = [
    [
      '@Plumo2:2020-07-26T15:15:43.411Z',
      '{"date":"2020-07-26T15:15:43.411Z","excercise":{"cycles":"10","id":1,"rest":"5","rounds":"2","title":"Custom Excercise 1"},"favourite":false,"level":2,"results":[["01:38","00:05"],["00:32",""]],"type":"Custom"}',
    ],
    [
      '@Plumo2:2020-06-12T15:15:43.411Z',
      '{"date":"2020-06-12T15:15:43.411Z","excercise":{"cycles":"11","id":2,"rest":"4","rounds":"3","title":"Endurance Excercise 2"},"favourite":false,"level":3,"results":[["01:38","00:15"],["00:32","00:15"],["02:38","00:15"]],"type":"Endurance"}',
    ],
    ['@Plumo2:2020-02-22T15:15:43.411Z', null],
    [
      '@Plumo2:2020-07-25T15:15:43.411Z',
      '{"date":"2020-07-25T15:15:43.411Z","excercise":{"cycles":"12","id":3,"rest":"5","rounds":"4","title":"Strength Excercise 3"},"favourite":true,"level":4,"results":[["01:38","01:00"]],"type":"Strength"}',
    ],
  ];

  afterEach(() => {
    cleanup();
    jest.clearAllMocks();
  });

  test('should render an empty screen when there are no activity IDs', async () => {
    getAsyncData.mockReturnValueOnce(null);

    await act(async () => {
      tree = create(
        <HistoryContextProvider idArray={[]}>
          <AllActivitiesScreen navigation={navigation} />
        </HistoryContextProvider>
      );
    });

    expect(tree).toMatchSnapshot();
  });

  test('should render an empty screen when there are no activities for the found activity IDs', async () => {
    getAsyncData.mockReturnValueOnce(activityIdArray);
    getMultiAsyncData.mockReturnValueOnce(null);

    await act(async () => {
      tree = create(
        <HistoryContextProvider idArray={[]}>
          <AllActivitiesScreen navigation={navigation} />
        </HistoryContextProvider>
      );
    });

    expect(tree).toMatchSnapshot();
  });

  test('should render all activities when present', async () => {
    getAsyncData.mockReturnValueOnce(activityIdArray);
    getMultiAsyncData.mockReturnValueOnce(activities);

    await act(async () => {
      tree = create(
        <HistoryContextProvider idArray={[]}>
          <AllActivitiesScreen navigation={navigation} />
        </HistoryContextProvider>
      );
    });

    expect(tree).toMatchSnapshot();
  });

  test('should render an empty screen if there is an error in favourited activity load', async () => {
    getAsyncData.mockReturnValueOnce(Promise.reject());
    console.log = jest.fn();

    await act(async () => {
      tree = create(
        <HistoryContextProvider idArray={[]}>
          <AllActivitiesScreen navigation={navigation} />
        </HistoryContextProvider>
      );
    });

    expect(console.log).toHaveBeenCalled();
    expect(tree).toMatchSnapshot();
  });

  describe('allActivitiesScreenOptions', () => {
    test('options', () => {
      const options = allActivitiesScreenOptions();
      expect(options).toMatchSnapshot();
    });

    test('headerTitle', () => {
      const headerTitle = allActivitiesScreenOptions().headerTitle({
        tintColor: 'white',
      });
      expect(headerTitle).toMatchSnapshot();
    });
  });
});
