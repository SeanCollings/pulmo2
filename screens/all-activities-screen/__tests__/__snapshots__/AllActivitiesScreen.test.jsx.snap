// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AllActivitiesScreen - unit test allActivitiesScreenOptions headerTitle 1`] = `
<Text
  style={
    Object {
      "color": "white",
      "fontSize": 20,
    }
  }
>
  All activities
</Text>
`;

exports[`AllActivitiesScreen - unit test allActivitiesScreenOptions options 1`] = `
Object {
  "cardStyleInterpolator": [Function],
  "headerTitle": [Function],
}
`;

exports[`AllActivitiesScreen - unit test should render all activities when present 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
    }
  }
>
  <RCTScrollView
    data={
      Array [
        Object {
          "date": "2020-07-26T15:15:43.411Z",
          "excercise": Object {
            "cycles": "10",
            "id": 1,
            "rest": "5",
            "rounds": "2",
            "title": "Custom Excercise 1",
          },
          "favourite": false,
          "level": 2,
          "results": Array [
            Array [
              "01:38",
              "00:05",
            ],
            Array [
              "00:32",
              "",
            ],
          ],
          "type": "Custom",
        },
        Object {
          "date": "2020-07-25T15:15:43.411Z",
          "excercise": Object {
            "cycles": "12",
            "id": 3,
            "rest": "5",
            "rounds": "4",
            "title": "Strength Excercise 3",
          },
          "favourite": true,
          "level": 4,
          "results": Array [
            Array [
              "01:38",
              "01:00",
            ],
          ],
          "type": "Strength",
        },
        Object {
          "date": "2020-06-12T15:15:43.411Z",
          "excercise": Object {
            "cycles": "11",
            "id": 2,
            "rest": "4",
            "rounds": "3",
            "title": "Endurance Excercise 2",
          },
          "favourite": false,
          "level": 3,
          "results": Array [
            Array [
              "01:38",
              "00:15",
            ],
            Array [
              "00:32",
              "00:15",
            ],
            Array [
              "02:38",
              "00:15",
            ],
          ],
          "type": "Endurance",
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={20}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReached={[Function]}
    onEndReachedThreshold={0.001}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={11}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "opacity": 1,
              "paddingTop": 5,
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "flexDirection": "row",
                "opacity": 1,
                "paddingBottom": 5,
                "paddingHorizontal": "10%",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                  "width": "20%",
                }
              }
            >
              <Text
                allowFontScaling={false}
                style={
                  Array [
                    Object {
                      "color": "#CCC",
                      "fontSize": 30,
                    },
                    undefined,
                    Object {
                      "fontFamily": "material-community",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                
              </Text>
            </View>
            <View
              style={
                Object {
                  "width": "80%",
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#002f56",
                    "fontFamily": "tit-regular",
                    "fontSize": 18,
                  }
                }
              >
                Custom Excercise 1
              </Text>
              <View
                style={
                  Object {
                    "flexDirection": "row",
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "#5bacf3",
                      "fontFamily": "tit-regular",
                      "fontSize": 16,
                    }
                  }
                >
                  Sun
                </Text>
                <Text
                  style={
                    Object {
                      "color": "#3e4b57",
                      "fontFamily": "tit-light",
                      "fontSize": 16,
                      "opacity": 1,
                    }
                  }
                >
                   26/07/2020 17:15
                </Text>
              </View>
              <View
                style={
                  Object {
                    "flexDirection": "row",
                    "justifyContent": "space-between",
                    "paddingRight": 10,
                    "width": "100%",
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "#3e4b57",
                      "fontFamily": "tit-light",
                      "fontSize": 16,
                      "opacity": 0.6,
                    }
                  }
                >
                  02:15
                </Text>
                <Text
                  style={
                    Object {
                      "color": "#3e4b57",
                      "fontFamily": "tit-light",
                      "fontSize": 16,
                      "opacity": 0.6,
                    }
                  }
                >
                  level: 
                  2
                </Text>
              </View>
            </View>
          </View>
          <View
            style={
              Object {
                "borderTopColor": "#CCC",
                "borderTopWidth": 1,
                "opacity": 0.6,
                "width": "80%",
              }
            }
          />
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "opacity": 1,
              "paddingTop": 5,
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "flexDirection": "row",
                "opacity": 1,
                "paddingBottom": 5,
                "paddingHorizontal": "10%",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                  "width": "20%",
                }
              }
            >
              <Text
                allowFontScaling={false}
                style={
                  Array [
                    Object {
                      "color": "#CCC",
                      "fontSize": 30,
                    },
                    undefined,
                    Object {
                      "fontFamily": "material-community",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                
              </Text>
            </View>
            <View
              style={
                Object {
                  "width": "80%",
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#002f56",
                    "fontFamily": "tit-regular",
                    "fontSize": 18,
                  }
                }
              >
                Strength Excercise 3
              </Text>
              <View
                style={
                  Object {
                    "flexDirection": "row",
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "#5bacf3",
                      "fontFamily": "tit-regular",
                      "fontSize": 16,
                    }
                  }
                >
                  Sat
                </Text>
                <Text
                  style={
                    Object {
                      "color": "#3e4b57",
                      "fontFamily": "tit-light",
                      "fontSize": 16,
                      "opacity": 1,
                    }
                  }
                >
                   25/07/2020 17:15
                </Text>
              </View>
              <View
                style={
                  Object {
                    "flexDirection": "row",
                    "justifyContent": "space-between",
                    "paddingRight": 10,
                    "width": "100%",
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "#3e4b57",
                      "fontFamily": "tit-light",
                      "fontSize": 16,
                      "opacity": 0.6,
                    }
                  }
                >
                  02:38
                </Text>
                <Text
                  style={
                    Object {
                      "color": "#3e4b57",
                      "fontFamily": "tit-light",
                      "fontSize": 16,
                      "opacity": 0.6,
                    }
                  }
                >
                  level: 
                  4
                </Text>
              </View>
            </View>
          </View>
          <View
            style={
              Object {
                "borderTopColor": "#CCC",
                "borderTopWidth": 1,
                "opacity": 0.6,
                "width": "80%",
              }
            }
          />
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "opacity": 1,
              "paddingTop": 5,
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "flexDirection": "row",
                "opacity": 1,
                "paddingBottom": 5,
                "paddingHorizontal": "10%",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                  "width": "20%",
                }
              }
            >
              <Text
                allowFontScaling={false}
                style={
                  Array [
                    Object {
                      "color": "#CCC",
                      "fontSize": 30,
                    },
                    undefined,
                    Object {
                      "fontFamily": "material-community",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                肋
              </Text>
            </View>
            <View
              style={
                Object {
                  "width": "80%",
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#002f56",
                    "fontFamily": "tit-regular",
                    "fontSize": 18,
                  }
                }
              >
                Endurance Excercise 2
              </Text>
              <View
                style={
                  Object {
                    "flexDirection": "row",
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "#5bacf3",
                      "fontFamily": "tit-regular",
                      "fontSize": 16,
                    }
                  }
                >
                  Fri
                </Text>
                <Text
                  style={
                    Object {
                      "color": "#3e4b57",
                      "fontFamily": "tit-light",
                      "fontSize": 16,
                      "opacity": 1,
                    }
                  }
                >
                   12/06/2020 17:15
                </Text>
              </View>
              <View
                style={
                  Object {
                    "flexDirection": "row",
                    "justifyContent": "space-between",
                    "paddingRight": 10,
                    "width": "100%",
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "#3e4b57",
                      "fontFamily": "tit-light",
                      "fontSize": 16,
                      "opacity": 0.6,
                    }
                  }
                >
                  05:33
                </Text>
                <Text
                  style={
                    Object {
                      "color": "#3e4b57",
                      "fontFamily": "tit-light",
                      "fontSize": 16,
                      "opacity": 0.6,
                    }
                  }
                >
                  level: 
                  3
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "height": 0,
          }
        }
      />
    </View>
  </RCTScrollView>
  <View
    style={
      Object {
        "alignSelf": "center",
        "backgroundColor": "white",
        "borderBottomWidth": 0,
        "borderColor": "#CCC",
        "borderTopLeftRadius": 50,
        "borderTopRightRadius": 50,
        "borderWidth": 1,
        "bottom": -100,
        "padding": 10,
        "position": "absolute",
      }
    }
  >
    <ActivityIndicator
      animating={true}
      color="#002f56"
      hidesWhenStopped={true}
      size="large"
    />
  </View>
</View>
`;

exports[`AllActivitiesScreen - unit test should render an empty screen if there is an error in favourited activity load 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
    }
  }
>
  <RCTScrollView
    data={Array []}
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={20}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReached={[Function]}
    onEndReachedThreshold={0.001}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={11}
  >
    <View />
  </RCTScrollView>
  <View
    style={
      Object {
        "alignSelf": "center",
        "flex": 1,
        "position": "absolute",
        "top": "50%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#3e4b57",
          "fontFamily": "tit-light",
          "fontSize": 16,
          "opacity": 1,
        }
      }
    >
      Saved activities are shown here.
    </Text>
  </View>
  <View
    style={
      Object {
        "alignSelf": "center",
        "backgroundColor": "white",
        "borderBottomWidth": 0,
        "borderColor": "#CCC",
        "borderTopLeftRadius": 50,
        "borderTopRightRadius": 50,
        "borderWidth": 1,
        "bottom": -100,
        "padding": 10,
        "position": "absolute",
      }
    }
  >
    <ActivityIndicator
      animating={true}
      color="#002f56"
      hidesWhenStopped={true}
      size="large"
    />
  </View>
</View>
`;

exports[`AllActivitiesScreen - unit test should render an empty screen when there are no activities for the found activity IDs 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
    }
  }
>
  <RCTScrollView
    data={Array []}
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={20}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReached={[Function]}
    onEndReachedThreshold={0.001}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={11}
  >
    <View />
  </RCTScrollView>
  <View
    style={
      Object {
        "alignSelf": "center",
        "flex": 1,
        "position": "absolute",
        "top": "50%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#3e4b57",
          "fontFamily": "tit-light",
          "fontSize": 16,
          "opacity": 1,
        }
      }
    >
      Saved activities are shown here.
    </Text>
  </View>
  <View
    style={
      Object {
        "alignSelf": "center",
        "backgroundColor": "white",
        "borderBottomWidth": 0,
        "borderColor": "#CCC",
        "borderTopLeftRadius": 50,
        "borderTopRightRadius": 50,
        "borderWidth": 1,
        "bottom": -100,
        "padding": 10,
        "position": "absolute",
      }
    }
  >
    <ActivityIndicator
      animating={true}
      color="#002f56"
      hidesWhenStopped={true}
      size="large"
    />
  </View>
</View>
`;

exports[`AllActivitiesScreen - unit test should render an empty screen when there are no activity IDs 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
    }
  }
>
  <RCTScrollView
    data={Array []}
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={20}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReached={[Function]}
    onEndReachedThreshold={0.001}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={11}
  >
    <View />
  </RCTScrollView>
  <View
    style={
      Object {
        "alignSelf": "center",
        "flex": 1,
        "position": "absolute",
        "top": "50%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#3e4b57",
          "fontFamily": "tit-light",
          "fontSize": 16,
          "opacity": 1,
        }
      }
    >
      Saved activities are shown here.
    </Text>
  </View>
  <View
    style={
      Object {
        "alignSelf": "center",
        "backgroundColor": "white",
        "borderBottomWidth": 0,
        "borderColor": "#CCC",
        "borderTopLeftRadius": 50,
        "borderTopRightRadius": 50,
        "borderWidth": 1,
        "bottom": -100,
        "padding": 10,
        "position": "absolute",
      }
    }
  >
    <ActivityIndicator
      animating={true}
      color="#002f56"
      hidesWhenStopped={true}
      size="large"
    />
  </View>
</View>
`;
